usr/sbin/so-clear-backlog

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+. /usr/sbin/so-common
+
+if [ -d /nsm/sensor_data ]; then
+	:
+else
+	echo "Not a sensor!  Exiting..."
+	exit 1;
+fi
+
+SKIP=0
+
+#########################################
+# Options
+#########################################
+usage()
+{
+cat <<EOF
+Security Onion Backlog Purge
+  Options:
+  -h         This message
+  -y         Skip interactive mode
+
+EOF
+}
+
+while getopts "h:y" OPTION
+do
+        case $OPTION in
+                h)
+                        usage
+                        exit 0
+                        ;;
+                y)
+                        SKIP=1
+                        ;;
+                *)
+                        usage
+                        exit 0
+                        ;;
+        esac
+done
+
+ENGINE=$(grep ENGINE /etc/nsm/securityonion.conf | cut -d= -f2)
+
+if [ $SKIP -ne 1 ] ; then
+	echo
+	echo "This script can be used to clear a backlog of unified2 files from"
+	echo "sensor data directories."
+	echo
+	echo "Please use this script with caution, as it will delete the following files:"
+	echo
+
+	if [ "$ENGINE" == "snort" ]; then
+		# Get count and listing of files
+		COUNT=$(ls -la /nsm/sensor_data/*/snort-*/snort.unified2.* | wc -l)
+		ls -la /nsm/sensor_data/*/snort-*/snort.unified2.*
+	else
+        	# Get count and listing of files
+		COUNT=$(ls -la /nsm/sensor_data/*/snort.unified2.* | wc -l)
+		ls -la /nsm/sensor_data/*/snort.unified2.*
+	fi
+	
+	echo
+	[[ ! -z $COUNT ]] && echo "$COUNT file(s) will be deleted."
+
+	echo
+	echo "If you wish to continue, please type YES and press the ENTER key"
+	echo
+	read INPUT
+        if [ "$INPUT" != "YES" ] ; then exit 0; fi
+fi
+
+# Check to see if Snort/Suricata is running -- if it is, stop it.
+e=$ENGINE
+if pgrep $e; then
+	nsm_sensor_stop --only-snort-alert
+fi
+
+# Check to see if barnyard2 is running -- if it is, stop it.
+if pgrep barnyard2; then
+	nsm_sensor_stop --only-barnyard2
+fi
+
+# If process is still not stopped, warn user and exit script.
+for i in $e Barnyard2 ; do
```

Comment:
Does `Barnyard2` need to be capitalized?

Reply:
That was merely for beginning the sentence with a capital letter, however I see that it actually cannot determine if the process is running or not with that, so I can revert to lowercase.